src/localization/online_localizer/online_localizer.cpp

```diff
@@ -326,94 +291,77 @@ bool OnlineLocalizer::predExists(const Node &node) const {
 
 void OnlineLocalizer::updateGraph(const Node &parent,
                                   const NodeSet &successors) {
-  if (successors.empty()) {
-    printf(
-        "[WARNING] No successors to add to the graph. May lead to disconnected "
-        "components\n");
-  }
-  // for every successor
+  LOG_IF(WARNING, successors.empty()) << "No successors to add to the graph. "
+                                         "May lead to disconnected components";
+
+  // For every successor
   // check if the child was visited before (The child was visited if there
   // exists a predecessor for it)
   // if yes, check if the proposed accumulated cost is smaller than existing one
   // if no set a pred for a child.
-  // printf("Number of successors %lu\n", successors.size());
   for (Node child : successors) {
     if (predExists(child)) {
       // child was visisted before
-      double prev_accCost = _accCosts[child.quId][child.refId];
+      double prev_accCost = accCosts_[child.quId][child.refId];
       double poss_accCost = child.idvCost + parent.accCost;
       if (poss_accCost < prev_accCost) {
-        // printf("[DEBUG][OnlineLocalizer] The child was visited before\n");
-        printf("[DEBUG][OnlineLocalizer] Possible accumulated cost is smaller "
-               "than previous one. Frontier should be updated\n");
-        printf(
-            "[BINGO] If you have reached this line, please contact Olga :)\n");
-        exit(0);
-        // throw 10;
+        LOG(INFO) << "Possible accumulated cost is smaller than previous one. "
+                     "Frontier should be updated";
+        LOG(FATAL)
+            << "[BINGO] If you have reached this line, please contact Olga :)";
```

Comment:
Maybe it's better to rephrase this to smth like "If this line is printed, please contact Olga" to avoid confusion of people reaching this line when reading the code 😉